### ISSUE TITLE
[CRITICAL] Can't boot without modifying boot commands

### ISSUE THREAD

**Issue by KaiserCalm** [ublue-os/bazzite#359]
I have reinstalled Bazzite OS with GNOME on Nvidia multiple times at this point and everything seems fine, it boots, but then after doing setup and restarting, the computer restarts twice and makes a new os tree, after that I can only boot by, in grub, manually adding:
`rd.driver.blacklist=nouveau modprobe.blacklist=nouveau nvidia-drm.modeset=1 nomodeset`
I really need some help.

**Comment by KyleGospo** [comment]
Are you using secure boot?

**Comment by KaiserCalm** [comment]
Yes. I have the keys enrolled though and disabling secure boot doesn't seem to change anything.

**Comment by KaiserCalm** [comment]
When I do boot into the de, `just update-system` and `just enroll-secure-boot-keys` are reported as not being present in the just file and I have to run `just update` and `just enroll-secure-boot-key` instead. For the key command, it returns that the key is already enrolled so it skips it.

**Comment by KaiserCalm** [comment]
This started happening after I ran `rpm-ostree update` trying to fix my other [issue](https://github.com/ublue-os/bazzite/issues/353#issuecomment-1732400701). I have been trying to fix this new issue by reinstalling the entire os multiple times now, but now the online installer does not get past Finished dracut-pre-pivot.service, so I used the offline installer with safe graphics for the most recent reinstall.

**Comment by vitalijb** [comment]
I ran into a similar problem, in my case after running 'rpm-ostree update' the following arguments were removed: rd.driver.blacklist=nouveau --append=modprobe.blacklist=nouveau --append=nvidia-drm.modeset=1 from grub.  Secure boot is disabled.

**Comment by SPECTREv333** [comment]
I have the same bug on bazzite-nvidia. Using `just nvidia-set-kargs` command does set the proper kernel args as i verified from grub, but the system does not complete the boot process and just reboots but with the kargs removed.
Secure boot is disabled.
Also checking `journalctl -b -1 -u ostree-finalize-staged.service` does not provide any useful information:
```set 25 22:19:23 x556ub systemd[1]: Finished ostree-finalize-staged.service - OSTree Finalize Staged Deployment.
set 25 22:19:33 x556ub systemd[1]: Stopping ostree-finalize-staged.service - OSTree Finalize Staged Deployment...
set 25 22:19:34 x556ub ostree[1305]: Finalizing staged deployment
set 25 22:19:35 x556ub ostree[1305]: Copying /etc changes: 25 modified, 3 removed, 87 added
set 25 22:19:35 x556ub ostree[1305]: Copying /etc changes: 25 modified, 3 removed, 87 added
set 25 22:19:37 x556ub ostree[1305]: Finalized deployment
set 25 22:19:47 x556ub ostree[1305]: Starting syncfs() for system root
set 25 22:19:47 x556ub ostree[1305]: Completed syncfs() for system root
set 25 22:19:47 x556ub ostree[1305]: Starting freeze/thaw cycle for system root
set 25 22:19:47 x556ub ostree[1305]: Completed freeze/thaw cycle for system root
set 25 22:19:47 x556ub ostree[1305]: Starting global sync()
set 25 22:19:47 x556ub ostree[1305]: Completed global sync()
set 25 22:19:47 x556ub ostree[1305]: Transaction complete; bootconfig swap: yes; bootversion: boot.0.1, deployment count change: 0
set 25 22:19:47 x556ub ostree[1305]: Transaction complete; bootconfig swap: yes; bootversion: boot.0.1, deployment count change: 0
set 25 22:19:47 x556ub ostree[1305]: Finished writing deployment
set 25 22:19:53 x556ub ostree[1305]: Cleanup complete
set 25 22:19:53 x556ub systemd[1]: ostree-finalize-staged.service: Deactivated successfully.
set 25 22:19:53 x556ub systemd[1]: Stopped ostree-finalize-staged.service - OSTree Finalize Staged Deployment.
set 25 22:19:53 x556ub systemd[1]: ostree-finalize-staged.service: Consumed 16.369s CPU time.
```
Same driver version works just fine on fedora workstation 38.

**Comment by KyleGospo** [comment]
Issue should be fixed as of the latest version

**Comment by vitalijb** [comment]
Hi there,
thanks a lot, the latest update seems to have addressed the issue, at least for me.

**Comment by SPECTREv333** [comment]
Hi, I can confirm the issue has been solved for me with 38-20230927 daily image. The fact that it fixed itself honestly felt like magic.
